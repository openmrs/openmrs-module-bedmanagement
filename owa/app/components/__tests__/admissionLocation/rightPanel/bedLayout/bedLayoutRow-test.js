import React from 'react';
import {shallow, mount} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';

import BedLayoutRow from 'components/admissionLocation/rightPanel/bedLayout/bedLayoutRow';
import admissionLocationFunctions from 'components/__mocks__/admissionLocationFunctions-mock';

const testProps = {
    admissionLocationFunctions: admissionLocationFunctions,
    loadAdmissionLocationLayout: (locationUuid) => jest.fn(),
    rowBeds: [null,
        {
            bedId: 9,
            bedNumber: '306-a',
            bedType: {
                description: 'deluxe',
                displayName: 'Double',
                id: 3
            },
            name: 'double',
            bedUuid: 'bb0a43db-d225-11e4-9c67-080027b662ec',
            columnNumber: 1,
            rowNumber: 2,
            status: 'AVAILABLE'
        },
        {
            bedId: null,
            bedNumber: null,
            bedType: null,
            bedUuid: null,
            columnNumber: 2,
            rowNumber: 2,
            status: null
        },
        {
            bedId: 10,
            bedNumber: '306-c',
            bedType: {
                name: 'deluxe',
                description: 'deluxe',
                displayName: 'DLX',
                id: 1
            },
            bedUuid: '4a83f079-6859-447a-aec7-e54b98ca8a73',
            columnNumber: 3,
            rowNumber: 2,
            status: 'AVAILABLE'
        }
    ]
};

describe('BedLayoutRow', () => {
    it('Should render bed layout properly', () =>{
        const bedLayoutRow = shallow(<BedLayoutRow admissionLocationFunctions={testProps.admissionLocationFunctions}
            row={2} layoutRow={3} layoutColumn={3} rowBeds={testProps.rowBeds} loadAdmissionLocationLayout={testProps.loadAdmissionLocationLayout} />);

        expect(bedLayoutRow.find('BedBlock').length).toBe(3);
        expect(shallowToJson(bedLayoutRow)).toMatchSnapshot();

        expect(bedLayoutRow.find('BedBlock').length).toBe(3);
        expect(shallowToJson(bedLayoutRow)).toMatchSnapshot();
    });
});
