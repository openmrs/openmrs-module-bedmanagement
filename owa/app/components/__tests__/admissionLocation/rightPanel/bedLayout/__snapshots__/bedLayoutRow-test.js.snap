// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BedLayoutRow Should render bed layout properly 1`] = `
<div
  className="bed-layout-row"
>
   
  <BedBlock
    admissionLocationFunctions={
      Object {
        "getActiveLocationUuid": [Function],
        "getAdmissionLocationByUuid": [Function],
        "getAdmissionLocations": [Function],
        "getBedTypes": [Function],
        "getChildAdmissionLocations": [Function],
        "getParentAdmissionLocation": [Function],
        "getState": [Function],
        "getVisitLocations": [Function],
        "notify": [Function],
        "reFetchAllAdmissionLocations": [Function],
        "setActiveLocationUuid": [Function],
        "setState": [Function],
      }
    }
    bed={
      Object {
        "bedId": 9,
        "bedNumber": "306-a",
        "bedType": Object {
          "description": "deluxe",
          "displayName": "Double",
          "id": 3,
        },
        "bedUuid": "bb0a43db-d225-11e4-9c67-080027b662ec",
        "columnNumber": 1,
        "name": "double",
        "rowNumber": 2,
        "status": "AVAILABLE",
      }
    }
    key="1"
    layoutColumn={3}
    layoutRow={3}
    loadAdmissionLocationLayout={[Function]}
  />
  <BedBlock
    admissionLocationFunctions={
      Object {
        "getActiveLocationUuid": [Function],
        "getAdmissionLocationByUuid": [Function],
        "getAdmissionLocations": [Function],
        "getBedTypes": [Function],
        "getChildAdmissionLocations": [Function],
        "getParentAdmissionLocation": [Function],
        "getState": [Function],
        "getVisitLocations": [Function],
        "notify": [Function],
        "reFetchAllAdmissionLocations": [Function],
        "setActiveLocationUuid": [Function],
        "setState": [Function],
      }
    }
    bed={
      Object {
        "bedId": null,
        "bedNumber": null,
        "bedType": null,
        "bedUuid": null,
        "columnNumber": 2,
        "rowNumber": 2,
        "status": null,
      }
    }
    key="2"
    layoutColumn={3}
    layoutRow={3}
    loadAdmissionLocationLayout={[Function]}
  />
  <BedBlock
    admissionLocationFunctions={
      Object {
        "getActiveLocationUuid": [Function],
        "getAdmissionLocationByUuid": [Function],
        "getAdmissionLocations": [Function],
        "getBedTypes": [Function],
        "getChildAdmissionLocations": [Function],
        "getParentAdmissionLocation": [Function],
        "getState": [Function],
        "getVisitLocations": [Function],
        "notify": [Function],
        "reFetchAllAdmissionLocations": [Function],
        "setActiveLocationUuid": [Function],
        "setState": [Function],
      }
    }
    bed={
      Object {
        "bedId": 10,
        "bedNumber": "306-c",
        "bedType": Object {
          "description": "deluxe",
          "displayName": "DLX",
          "id": 1,
          "name": "deluxe",
        },
        "bedUuid": "4a83f079-6859-447a-aec7-e54b98ca8a73",
        "columnNumber": 3,
        "rowNumber": 2,
        "status": "AVAILABLE",
      }
    }
    key="3"
    layoutColumn={3}
    layoutRow={3}
    loadAdmissionLocationLayout={[Function]}
  />
</div>
`;

exports[`BedLayoutRow Should render bed layout properly 2`] = `
<div
  className="bed-layout-row"
>
   
  <BedBlock
    admissionLocationFunctions={
      Object {
        "getActiveLocationUuid": [Function],
        "getAdmissionLocationByUuid": [Function],
        "getAdmissionLocations": [Function],
        "getBedTypes": [Function],
        "getChildAdmissionLocations": [Function],
        "getParentAdmissionLocation": [Function],
        "getState": [Function],
        "getVisitLocations": [Function],
        "notify": [Function],
        "reFetchAllAdmissionLocations": [Function],
        "setActiveLocationUuid": [Function],
        "setState": [Function],
      }
    }
    bed={
      Object {
        "bedId": 9,
        "bedNumber": "306-a",
        "bedType": Object {
          "description": "deluxe",
          "displayName": "Double",
          "id": 3,
        },
        "bedUuid": "bb0a43db-d225-11e4-9c67-080027b662ec",
        "columnNumber": 1,
        "name": "double",
        "rowNumber": 2,
        "status": "AVAILABLE",
      }
    }
    key="1"
    layoutColumn={3}
    layoutRow={3}
    loadAdmissionLocationLayout={[Function]}
  />
  <BedBlock
    admissionLocationFunctions={
      Object {
        "getActiveLocationUuid": [Function],
        "getAdmissionLocationByUuid": [Function],
        "getAdmissionLocations": [Function],
        "getBedTypes": [Function],
        "getChildAdmissionLocations": [Function],
        "getParentAdmissionLocation": [Function],
        "getState": [Function],
        "getVisitLocations": [Function],
        "notify": [Function],
        "reFetchAllAdmissionLocations": [Function],
        "setActiveLocationUuid": [Function],
        "setState": [Function],
      }
    }
    bed={
      Object {
        "bedId": null,
        "bedNumber": null,
        "bedType": null,
        "bedUuid": null,
        "columnNumber": 2,
        "rowNumber": 2,
        "status": null,
      }
    }
    key="2"
    layoutColumn={3}
    layoutRow={3}
    loadAdmissionLocationLayout={[Function]}
  />
  <BedBlock
    admissionLocationFunctions={
      Object {
        "getActiveLocationUuid": [Function],
        "getAdmissionLocationByUuid": [Function],
        "getAdmissionLocations": [Function],
        "getBedTypes": [Function],
        "getChildAdmissionLocations": [Function],
        "getParentAdmissionLocation": [Function],
        "getState": [Function],
        "getVisitLocations": [Function],
        "notify": [Function],
        "reFetchAllAdmissionLocations": [Function],
        "setActiveLocationUuid": [Function],
        "setState": [Function],
      }
    }
    bed={
      Object {
        "bedId": 10,
        "bedNumber": "306-c",
        "bedType": Object {
          "description": "deluxe",
          "displayName": "DLX",
          "id": 1,
          "name": "deluxe",
        },
        "bedUuid": "4a83f079-6859-447a-aec7-e54b98ca8a73",
        "columnNumber": 3,
        "rowNumber": 2,
        "status": "AVAILABLE",
      }
    }
    key="3"
    layoutColumn={3}
    layoutRow={3}
    loadAdmissionLocationLayout={[Function]}
  />
</div>
`;
